arangod/Cache/Cache.h

```diff
@@ -251,21 +266,52 @@ class Cache : public std::enable_shared_from_this<Cache> {
                              std::unique_ptr<Table::Subtable> targets,
                              Table& newTable) = 0;
 
-  static constexpr std::uint64_t findStatsCapacity = 16384;
+  void ensureFindStats();
 
   basics::ReadWriteSpinLock _taskLock;
   std::atomic<bool> _shutdown;
 
-  std::unique_ptr<StatBuffer> _findStats;
-  mutable basics::SharedCounter<64> _findHits;
-  mutable basics::SharedCounter<64> _findMisses;
-
   // allow communication with manager
   Manager* _manager;
   std::uint64_t const _id;
   Metadata _metadata;
 
+  // local buffer for tracking allocations/deallocations by this cache.
+  // this value will be modified locally until the value of |_memoryUsageDiff|
+  // exceeds the reporting granularity threshold.
+  // once the threshold is exceeded, the current value of _memoryUsageDiff will
+  // be reported to the manager, and the value of _memoryUsageDiff will be reset
+  // to zero. that means caches may report their memory usage in a delayed
+  // fashion. additionally, the last kMemoryReportGranularity bytes that were
+  // already used by the cache may not have been reported to the manager. these
+  // bytes will only be missing temporarily though. the missing bytes can be
+  // reported later, when there are additional allocations or deallocations,
+  // or when the cache gets destroyed. they will only be not be missing
+  // completely.
+  std::atomic<std::int64_t> _memoryUsageDiff;
+
+  // this struct is allocated on the heap only lazily
+  struct FindStats {
+    mutable basics::SharedCounter<64> findHits;
+    mutable basics::SharedCounter<64> findMisses;
+    std::unique_ptr<StatBuffer> findStats;
+  };
+
+  // this will only flip from false to true, when the eviction stats
+  // are created
+  std::atomic<bool> _haveFindStats;
+
+  bool const _enableWindowedStats;
+
+  // used to create eviction stats under the lock (so that only a single
+  // thread creates them)
+  std::mutex _findStatsLock;
+  // this struct is allocated lazily, when the _findStats are first written to
+  std::unique_ptr<FindStats> _findStats;
```

Comment:
https://github.com/abseil/abseil-cpp/blob/master/absl/base/call_once.h
We could use this to decrease sizeof and speedup it.

Reply:
I understand that absl::call_once() would help us in the majority of cases, but it seems it does not provide a way to query the status of the `once_flag` from the outside.

This would be necessary in `Cache::hitRates()`, where we check if we have created findStats already, but do **not** want to create them if not. IMO `absl::call_once()` does not allow this.